staging/src/k8s.io/code-generator/cmd/client-gen/args/args.go

```diff
@@ -78,6 +91,32 @@ func (ca *CustomArgs) AddFlags(fs *pflag.FlagSet) {
 	pflag.StringVarP(&ca.ClientsetName, "clientset-name", "n", "internalclientset", "the name of the generated clientset package.")
 	pflag.StringVarP(&ca.ClientsetAPIPath, "clientset-api-path", "", "", "the value of default API path.")
 	pflag.StringVar(&ca.ClientsetOutputPath, "clientset-path", "k8s.io/kubernetes/pkg/client/clientset_generated/", "the generated clientset will be output to <clientset-path>/<clientset-name>.")
+	pflag.StringVarP(&ca.NamerExceptionsFile, "namer-exceptions-file", "", "", "a YAML file that contains namer exceptions.")
 	pflag.BoolVar(&ca.ClientsetOnly, "clientset-only", false, "when set, client-gen only generates the clientset shell, without generating the individual typed clients")
 	pflag.BoolVar(&ca.FakeClient, "fake-clientset", true, "when set, client-gen will generate the fake clientset that can be used in tests")
 }
+
+var defaultPluralizationException = map[string]string{
+	"Endpoints": "Endpoints",
+}
+
+// LoadNamerExceptions loads the namer exception from provided YAML file.
+func LoadNamerExceptions(fileName string) (NamerExceptions, error) {
+	var result NamerExceptions
+	if len(fileName) == 0 {
```

Comment:
this ==0 check belongs into main.go IMO

Reply:
@sttts wanted to avoid initialization of NamerExceptions there...